src/server/pkg/storage/tracker/gc.go

```diff
@@ -0,0 +1,93 @@
+package tracker
+
+import (
+	"context"
+	"time"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Deleter interface {
+	Delete(ctx context.Context, id string) error
+}
+
+type DeleterMux func(string) Deleter
+
+func (dm DeleterMux) Delete(ctx context.Context, id string) error {
+	deleter := dm(id)
+	if deleter == nil {
+		return errors.Errorf("deleter mux does not have deleter for (%s)", id)
+	}
+	return deleter.Delete(ctx, id)
+}
+
+type GarbageCollector struct {
+	tracker Tracker
+	period  time.Duration
+	deleter Deleter
+}
+
+func NewGarbageCollector(tracker Tracker, period time.Duration, deleter Deleter) *GarbageCollector {
+	return &GarbageCollector{
+		tracker: tracker,
+		period:  period,
+		deleter: deleter,
+	}
+}
+
+func (GarbageCollector *GarbageCollector) Run(ctx context.Context) error {
```

Comment:
We don't need to spell out `GarbageCollector` for the local variables, just the struct type name. The general pattern I like to follow, and I think that you usually follow, is a letter per word in the struct type name.